add input calib provenance keywords

sbailey opened the pull request:
This PR fixes #997 by adding header keywords to track which calibs were used.  Example outputs in /global/cfs/cdirs/desi/users/sjbailey/spectro/redux/test where 20200315/00055611 was processed with default calibs from $DESI_SPECTRO_CALIB, while 20201220/00069020 was processed with CCD calibs from $DESI_SPECTRO_CALIB but PSF and fiberflat from nightly calibs.

For low-level CCD calibs, I used DEPNAMnn/DEPVERnn keyword pairs to allow longer keyword names, e.g.
```
DEPNAM00= 'DESI_SPECTRO_CALIB'                                                  
DEPVER00= '/global/cfs/cdirs/desi/spectro/desi_spectro_calib/trunk'             
DEPNAM01= 'CCD_CALIB_BIAS'                                                      
DEPVER01= 'SPCALIB/ccd/bias-sm10-r1-20201115.fits.gz'                           
DEPNAM02= 'CCD_CALIB_DARK'                                                      
DEPVER02= 'SPCALIB/ccd/dark-sm10-r1-20201115.fits.gz'                           
DEPNAM03= 'CCD_CALIB_MASK'                                                      
DEPVER03= 'SPCALIB/ccd/pixmask-sm10-r-20201014.fits.gz'                         
```
while for downstream steps I used keywords like IN_PSF, IN_IMG, etc.  To keep filenames short enough to not require CONTINUE keywords, I also shorten them by replacing prefixes
  * $DESI_SPECTRO_CALIB -> SPCALIB
  * $DESI_SPECTRO_REDUX/$SPECPROD -> SPECPROD

@julienguy please review for including in 20.12 / blanc.

julienguy reviewed (commented):
Running on one exposure 20201216/00068289 cameras b0,r0,z0, I see
 * two psfs mentioned, one used for preproc (model variance), one for extraction. It's fine for me but maybe this could be confusing. 
```
desi_header -i frame-r0-00068289.fits | grep -i psf 
DEPNAM05= 'CCD_CALIB_PSF'                                                       
DEPVER05= 'SPCALIB/spec/sm4/psfnight-r0-20201214.fits'                          
IN_PSF  = 'SPECPROD/exposures/20201216/00068289/psf-r0-00068289.fits' / Input sp
```
 * in the sframe, the fiber flat used in preproc but not the one actually used to flatfield the exposure

desi_header -i sframe-r0-00068289.fits | grep -i flat
```
DEPNAM04= 'CCD_CALIB_PIXFLAT'                                                   
DEPVER04= 'SPCALIB/ccd/pixflat-sm4-r.fits.gz'                                   
DEPNAM06= 'CCD_CALIB_FIBERFLAT'                                                 
DEPVER06= 'SPCALIB/spec/sm4/fiberflatnight-r0-20201214.fits'  
```

 - For the cframe, we have the correct flat mentioned. So the action might be simply add the mention of flat in the sframe header.

desi_header -i cframe-r0-00068289.fits | grep -i flat
```
DEPNAM04= 'CCD_CALIB_PIXFLAT'                                                   
DEPVER04= 'SPCALIB/ccd/pixflat-sm4-r.fits.gz'                                   
DEPNAM06= 'CCD_CALIB_FIBERFLAT'                                                 
DEPVER06= 'SPCALIB/spec/sm4/fiberflatnight-r0-20201214.fits'
FIBERFLT= 'SPECPROD/calibnight/20201216/fiberflatnight-r0-20201216.fits'
```

I checked the CCD preproc calib files are all there. This is great.

sbailey commented:
Indeed, there are two PSFs listed separately in the frame files because the PSF used for the CCD noise model could be different from the PSF used for extracting that image.  I think listing both is a feature not a bug.  As one more bonus, I added a 3rd "ORIG_PSF" to the frame output, tracking which PSF was used before trace-shifting.  Summarizing:
  * CCD_CALIB_PSF: PSF used for CCD noise modeling
  * IN_PSF: the PSF used for the spectroperfectionism extractions; typically the PSF in the same directory as the frame file (i.e. not necessarily very informative) 
  * ORIG_PSF: the pre-traceshifted PSF that was adjusted to become the IN_PSF.  Most nights this should be the psfnight made from arcs on that night, with a fallback to a DESI_SPECTRO_CALIB default PSF.  Adding this keyword allows you to tell from the frame file itself whether its PSF came from a nightly PSF vs. a default PSF without having to track back through the traceshifted IN_PSF.

But it was a bug that I had forgotten to add the input sky and fiberflat files to the sframe outputs.  I added that.

I retested with a full desi_proc end-to-end after making these updates, but now I see that GitHub unit tests are unhappy so I'll go check that...

sbailey commented:
Discussed with @julienguy; merging now.
